### ISSUE TITLE
Ability to define Children Prototype

### ISSUE THREAD

**Issue by futape** [neos/fusion-afx#8]
There should be the ability to define the prototype a tag's children a nested into (`Neos.Fusion:Array` by default).
Currently it's not possible to construct arbitrary data in AFX. For example it is not possible to pass a prop containing a `Neos.Fusion:RawArray` value like this:

```
afx`
  <Vendor.Site:Foo @children="bar">
    <Vendor.Site:Foo.Header @key="header" />
    <Vendor.Site:Foo.Body @key="body" />
  </Vendor.Site:Foo @children="bar">
`
```

Instead `Vendor.Site:Foo`'s `bar` prop would contain:

```
bar = Neos.Fusion:Array {
  header = Vendor.Site:Foo.Header
  body = Vendor.Site:Foo.Body
}
```

I suggest a new meta property like `@childrenType` to controle the prototype used, which defaults to `Neos.Fusion:Array` if not set. Also the result should **alyways** be a value of that type, even if only one child exists, so that the user can always rely on his configuration.

The above example would be written as:

```
afx`
  <Vendor.Site:Foo @children="bar" @childrenType="Neos.Fusion:RawArray">
    <Vendor.Site:Foo.Header @key="header" />
    <Vendor.Site:Foo.Body @key="body" />
  </Vendor.Site:Foo @children="bar">
`
```

and would be transpiled to:

```
bar = Neos.Fusion:RawArray {
  header = Vendor.Site:Foo.Header
  body = Vendor.Site:Foo.Body
}
```

**Comment by mficzel** [comment]
@futape i suggest to close this in favor of the approach in pr https://github.com/neos/fusion-afx/pull/12 that directly addresses the need to render multiple afx children into a fusion property.

With @path your scenario would look like this: 

afx`
  <Vendor.Site:Foo>
      <Neos.Fusion:RawArray @path="bar>
         <Vendor.Site:Foo.Header @path="header" />
         <Vendor.Site:Foo.Body @path="body" />
     </Neos.Fusion:RawArray>
  </Vendor.Site:Foo>
`

**Comment by jonnitto** [comment]
Hi @futape 
Is it ok for you if we close your issue?

**Comment by mficzel** [comment]
Closed in favor of pr #12. Please reopen the solution there is not enough.
